src/Simulation/EntryPointDriver/Driver.cs

```diff
@@ -241,146 +291,220 @@ private void AddOptionIfAvailable<T>(Command command, OptionInfo<T> option)
+
+            return (command, validators);
+        }
 
         /// <summary>
-        /// The shots option.
+        /// Creates a sub command specific to the given entry point for the submit command.
         /// </summary>
-        internal static readonly OptionInfo<int> ShotsOption = new OptionInfo<int>(
-            ImmutableList.Create("--shots"),
-            500,
-            "The number of times the program is executed on the target machine.",
-            validator: result =>
-                int.TryParse(result.Tokens.SingleOrDefault()?.Value, out var value) && value <= 0
-                    ? "The number of shots must be a positive number."
-                    : default);
+        /// <param name="entryPoint">The entry point to make a command for.</param>
+        /// <returns>Tuple with the command corresponding to the given entry point and the validators for that command.</returns>
+        private (Command, IEnumerable<ValidateSymbol<CommandResult>>) CreateSubmitEntryPointCommand(IEntryPoint entryPoint)
+        {
+            var validators = Enumerable.Empty<ValidateSymbol<CommandResult>>();
+
+            var command = new Command(entryPoint.Name, entryPoint.Summary)
+            {
+                Handler = CommandHandler.Create((ParseResult parseResult, AzureSettings settings) => this.Submit(parseResult, settings, entryPoint))
+            };
+            foreach (var option in entryPoint.Options)
+            {
+                command.AddOption(option);
+            }
+
+            validators = validators.Concat(AddOptionIfAvailable(command, SubscriptionOption));
+            validators = validators.Concat(AddOptionIfAvailable(command, ResourceGroupOption));
+            validators = validators.Concat(AddOptionIfAvailable(command, WorkspaceOption));
+            validators = validators.Concat(AddOptionIfAvailable(command, this.CreateTargetOption(entryPoint)));
+            validators = validators.Concat(AddOptionIfAvailable(command, StorageOption));
+            validators = validators.Concat(AddOptionIfAvailable(command, AadTokenOption));
+            validators = validators.Concat(AddOptionIfAvailable(command, BaseUriOption));
+            validators = validators.Concat(AddOptionIfAvailable(command, LocationOption));
+            validators = validators.Concat(AddOptionIfAvailable(command, JobNameOption));
+            validators = validators.Concat(AddOptionIfAvailable(command, ShotsOption));
+            validators = validators.Concat(AddOptionIfAvailable(command, OutputOption));
+            validators = validators.Concat(AddOptionIfAvailable(command, DryRunOption));
+            validators = validators.Concat(AddOptionIfAvailable(command, VerboseOption));
+            validators = validators.Concat(MarkOptionsAsMutuallyExclusive(
+                command,
+                new[] { BaseUriOption.Aliases.First(), LocationOption.Aliases.First() }));
```

Comment:
nit: You can chain the `.Concat`s instead of reassigning `validators` each time.

Reply:
done